addons/ui/functions/fnc_setAdvancedElement.sqf

```diff
@@ -18,21 +18,29 @@
  * Public: No
  */
 
-params ["_element", "_show", ["_showHint", false, [true]], ["_force", false, [true]] ];
+params ["_element", "_show", ["_showHint", false, [true]], ["_force", false, [true]]];
 
 private _cachedElement = GVAR(configCache) getVariable _element;
-if (isNil "_cachedElement") exitWith {};
+if (isNil "_cachedElement") exitWith {TRACE_1("nil element",_this)};
 
 if (!_force && {!GVAR(allowSelectiveUI)}) exitWith {
+    TRACE_1("not allowed",_this);
     [LSTRING(Disallowed), 2] call EFUNC(common,displayTextStructured);
     false
 };
 
 _cachedElement params ["_idd", "_elements", "_location", "_conditions"];
 
 // Exit if main vehicle type condition not fitting
-private _canUseWeapon = ACE_player call CBA_fnc_canUseWeapon;
-if ((_canUseWeapon && {_location == 2}) || {!_canUseWeapon && {_location == 1}}) exitWith {false};
+private _cargoIndex = vehicle ACE_player getCargoIndex ACE_player; // nil if not in vehicle
+private _isOnFootOrInCargo = isNil "_cargoIndex" || {_cargoIndex > -1};
```

Comment:
```suggestion
private _isOnFootOrInCargo = isNil "_cargoIndex" || {_cargoIndex > -1} || {ACE_player call CBA_fnc_canUseWeapon};
```
Need this to handle non-cargo FFV turrets.
Example with merkel tank with commander gun,
inside it shows ammo (gunner ammo) fine
turn out and player's gun ammo shows, and exiting while in this state it will continue to show dismounted.

While turned out in turret::
ACE_player call CBA_fnc_canUseWeapon = false
vehicle ACE_player getCargoIndex ACE_player = -1

Could possibly also use `isTurnedOut player`, but cba func isn't that expensive